### PR TITLE
Jr starting on UI

### PR DESCRIPTION
**Pull request by jb520**
Connected bootstrap and added navbar with conditional view for signed-in/signed-out users. Replaced auto-generated notices with new formatting in _flashmessages.html.erb. Edited sample_data rake task for easier testing.

### CODE REVIEW

**Review by MayaS1111 — APPROVED** [review]
Great job, looks good

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,3 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
```
Comment (JoseMDO) [inline app/controllers/application_controller.rb:2]: Nice!

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -6,11 +6,26 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
+    <%= render partial: "shared/cdn_assets" %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:9]: great job adding your cdn assets to a partial! This helps keep the app layout file clean :)

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -6,11 +6,26 @@
...
+
     <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
     <%= javascript_importmap_tags %>
   </head>
 
   <body>
-    <%= yield %>
+
+    <%= render partial: "shared/navbar" %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:17]: same as above

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -6,11 +6,26 @@
...
+
+    <div class="container">
+      <% if notice.present? %>
+        <%= render partial: "shared/flash", locals: { message: notice, css_class: "success" } %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:21]: same as above

File: app/views/shared/_navbar.html.erb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      <span class="navbar-toggler-icon"></span>
+    </button>
+    <div class="collapse navbar-collapse" id="navbarSupportedContent">
+      <ul class="navbar-nav mr-auto">
+        <li class="nav-item dropdown">
+          <a class="nav-link dropdown-toggle" href="#" role="button" data-bs-toggle="dropdown" aria-expanded="false">
+            Resources
+          </a>
+          <div class="dropdown-menu">
+            <a class="dropdown-item" href="/photos">Photos</a>
```
Comment (JoseMDO) [inline app/views/shared/_navbar.html.erb:15]: remember we can start using helper methods now like link_to

File: app/views/shared/_navbar.html.erb [hunk 6]
```diff
@@ -0,0 +1,42 @@
+<nav class="navbar navbar-expand-lg bg-light mb-4">
+  <div class="container">
+    <!--<a class="navbar-brand" href="#">Navbar</a>-->
```
Comment (JoseMDO) [inline app/views/shared/_navbar.html.erb:3]: try to delete uncommented code if it is no longer being used

File: lib/tasks/dev.rake [hunk 7]
```diff
@@ -11,9 +11,23 @@ task({ :sample_data => :environment }) do
     User.destroy_all
   end
 
+  usernames = Array.new { Faker::Name.first_name }
+
+  usernames << "alice"
+  usernames << "bob"
+
+  usernames.each do |username|
```
Comment (JoseMDO) [inline lib/tasks/dev.rake:19]: great job on sample data!

**Review by JoseMDO — APPROVED** [review]
Great job Jennifer, everything looks great!

**Pull request merged**